### PR TITLE
Fix and extend ASP.NET integration tests

### PR DESCRIPTION
**Pull request by andrewlock**
Fixes ASP.NET MVC 4/5 integration tests being skipped (I think) since https://github.com/DataDog/dd-trace-dotnet/commit/fadfcb4135cbb63d3305808b1cfb2a7ec966d97e due to missing `IXunitSerializable` properties (could be run locally, but seemed to erroneously pass in CI)

Adds additional test cases for route parameter handling - optional parameters, bad values etc. Using conventional and attribute routing. 

@DataDog/apm-dotnet

### CODE REVIEW

File: test/Datadog.Trace.ClrProfiler.IntegrationTests/AspNet/AspNetMvc5Tests.cs [hunk 1]
```diff
@@ -2,15 +2,14 @@
...
-using System.Collections.Generic;
 using System.Net;
 using System.Threading.Tasks;
 using Xunit;
 using Xunit.Abstractions;
 
 namespace Datadog.Trace.ClrProfiler.IntegrationTests
 {
-    [Collection("IisTests")]
+    [Collection(nameof(IisFixture))]
```
Comment (lucaspimentel) [inline test/Datadog.Trace.ClrProfiler.IntegrationTests/AspNet/AspNetMvc5Tests.cs:12]: The MVC 4 tests use `[Collection("IisTests")]` so these will have a different collection name. Was that intentional?

File: test/Datadog.Trace.ClrProfiler.IntegrationTests/Helpers/SerializableDictionary.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+using Xunit.Abstractions;
+
+namespace Datadog.Trace.ClrProfiler.IntegrationTests
+{
+    public class SerializableDictionary : IXunitSerializable, IEnumerable<KeyValuePair<string, string>>
```
Comment (lucaspimentel) [inline test/Datadog.Trace.ClrProfiler.IntegrationTests/Helpers/SerializableDictionary.cs:8]: I'm surprised that xUnit does not support serializing ` Dictionary<string, string>` out of the box. 🤔

File: test/Datadog.Trace.ClrProfiler.IntegrationTests/AspNet/AspNetMvc5Tests.cs (hunk 1 above)
Comment (andrewlock) [reply test/Datadog.Trace.ClrProfiler.IntegrationTests/AspNet/AspNetMvc5Tests.cs:12]: Oops, no, that was something else

File: test/Datadog.Trace.ClrProfiler.IntegrationTests/Helpers/SerializableDictionary.cs (hunk 2 above)
Comment (andrewlock) [reply test/Datadog.Trace.ClrProfiler.IntegrationTests/Helpers/SerializableDictionary.cs:8]: Yeah, I need to double check, as I initially assumed the dictionary was the problem, but later found that `HttpStatusCode` was an issue. But given even `Guid` is not serializable by default, I strongly suspect the dictionary is not either 😉

File: test/Datadog.Trace.ClrProfiler.IntegrationTests/AspNet/AspNetMvc5Tests.cs (hunk 1 above)
Comment (andrewlock) [reply test/Datadog.Trace.ClrProfiler.IntegrationTests/AspNet/AspNetMvc5Tests.cs]: Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/1246/commits/7d3b76f5a67a58d6c14e758097cb28eae2f4b333

File: test/Datadog.Trace.ClrProfiler.IntegrationTests/Helpers/SerializableDictionary.cs (hunk 2 above)
Comment (andrewlock) [reply test/Datadog.Trace.ClrProfiler.IntegrationTests/Helpers/SerializableDictionary.cs:8]: Yep, confirmed, you can't use `Dictionary<string, string>` 🙄

Comment (lucaspimentel) [reply test/Datadog.Trace.ClrProfiler.IntegrationTests/Helpers/SerializableDictionary.cs:8]: Thanks for checking!

**Review by lucaspimentel — APPROVED** [review]

**Review by zacharycmontoya — APPROVED** [review]

**Pull request merged**